modules/doublet-detection/resources/usr/bin/run_scdblfinder.R

```diff
@@ -0,0 +1,167 @@
+    dplyr::select(dplyr::all_of(columns_to_keep))
+
+  return(result_df)
+}
+
+
+# Parse options --------
+
+option_list <- list(
+  make_option(
+    "--input_sce_file",
+    type = "character",
+    help = "Path to input SCE file in RDS format."
+  ),
+  make_option(
+    "--output_file",
+    type = "character",
+    default = "scdblfinder_results.tsv",
+    help = "Path to output TSV file with doublet inferences."
+  ),
+  make_option(
+    "--cores",
+    type = "integer",
+    default = 4,
+    help = "Number of cores to use during scDblFinder inference."
+  ),
+  make_option(
+    "--benchmark",
+    action = "store_true",
+    default = FALSE,
+    help = "Whether the script is being run as part of the doublet benchmarking analysis. If this flag is invoked, the `cxds_score` column will be included in the output TSV along with the `barcodes`, `score`, and `class` columns that are always included."
+  ),
+  make_option(
+    "--sample_var",
+    type = "character",
+    help = "If multiple samples are present in the SCE, the colData column name with sample ids. This option should be used for merged objects."
+  ),
+  make_option(
+    "--random_seed",
+    type = "integer",
+    default = 2024,
+    help = "Random seed. Default is 2024."
+  )
+)
+opts <- parse_args(OptionParser(option_list = option_list))
+
+# Check input arguments and set up files, directories ------
+if (!file.exists(opts$input_sce_file)) {
+  glue::glue("Could not find input SCE file, expected at: `{opts$input_sce_file}`.")
+}
+if (!stringr::str_ends(opts$output_file, ".tsv")) {
+  stop("The output TSV file must end in .tsv.")
+}
+
+fs::dir_create(dirname(opts$output_file)) # create output directory as needed
+set.seed(opts$random_seed)
+
+# Detect doublets and export TSV file with inferences -----
+
+# Check the number of cells to see if scDblFinder can be run
+cell_threshold <- 10
+
+sce <- readRDS(opts$input_sce_file)
+ncells <- ncol(sce)
+
+if (ncells < cell_threshold) {
+  warning(
+    glue::glue(
+      "The provided SingleCellExperiment object only has {ncells} droplets, which is fewer than {cell_threshold} so `scDblFinder` will not be run.
+       An output TSV file will still be produced, but it will be populated with `NA` values."
+    )
+  )
+  na_df <- data.frame(
+    barcodes = colnames(sce),
+    score = NA,
+    class = NA
+  )
+  if (opts$benchmark) {
+    na_df$cxds_score <- NA
+  }
+  readr::write_tsv(na_df, opts$output_file)
+} else {
+  keep_columns <- c(
+    "barcodes",
+    "score",
+    "class"
+  )
+  if (opts$benchmark) {
+    keep_columns <- c(keep_columns, "cxds_score")
+  }
+
+  run_scdblfinder(
+    sce,
+    cores = opts$cores,
+    # only used if there are multiple samples, e.g. if this is processing a merged object
+    sample_var = opts$sample_var, # will be NULL if not provided as input argument
+    random_seed = opts$random_seed,
+    columns_to_keep = keep_columns
+  ) |>
+    readr::write_tsv(opts$output_file)
```

Comment:
This is a minor suggestion, mostly because I don't love piping directly to output, especially when the using a custom function with a potentially complex output.  But making this change here would require a change in `scpca-analysis` and an updated hash for the copy, so it probably isn't worth it at this stage.  
```suggestion
  scdblfinder_results <- run_scdblfinder(
    sce,
    cores = opts$cores,
    # only used if there are multiple samples, e.g. if this is processing a merged object
    sample_var = opts$sample_var, # will be NULL if not provided as input argument
    random_seed = opts$random_seed,
    columns_to_keep = keep_columns
  )
  readr::write_tsv(scdblfinder_results, opts$output_file)
```

Reply:
ignoring per previous comment